### PR TITLE
Pin requirements to specific versions and start removing python 2 support

### LINKED ISSUE #486: Disable the Python 2 Travis tests
As mentioned in https://github.com/mozilla/OpenWPM/issues/466#issuecomment-522209540

### LINKED ISSUE #499: Pin versions for all pip-installed dependencies
We've had several issues randomly pop up due to changes in our underlying dependencies. These include #439 and #498. We should really pin all of the pip-installed requirements to specific versions. One approach would be to switch over to using `pipenv` as described in #205.

[This post](https://hynek.me/articles/python-app-deps-2018/) gives a good comparison between our options (at least as of ~1 year ago). The `pip-tools` suggestion at the end of the post looks reasonable. As stated there, the downside is the lack of support for conditional dependencies. This seems acceptable as we've already decided to drop support for python 2 (see #485).

### PR DESCRIPTION
**Pull request by englehardt**
(~Includes #498, but will be rebased once that is merged)~ DONE

This PR does three things:
* Uses `pip-compile` from `pip-tools` to pin requirements to specific versions. Fixes #499.
* Disables python 2.7 in the tests and removes python 2 support from a number of the install files. We previous decided to drop python 2 support in #466. I'm starting this in this PR from the reasons described in #499, namely incompatibilities between python 2 and python 3 pinned dependencies. Fixes #486.
* Removes the test that checks if python packages are installed for the reasons described in the commit message for be771c839941aa3ec003a90360c9c2a979a550db.

### CODE REVIEW

File: automation/DataAggregator/S3Aggregator.py [hunk 1]
```diff
@@ -221,7 +221,6 @@ def _send_to_s3(self, force=False):
                     pq.write_to_dataset(
                         table, self._s3_bucket_uri % table_name,
                         filesystem=self._fs,
-                        preserve_index=False,
```
[the remark this answers is not in the document]
Comment (englehardt) [reply automation/DataAggregator/S3Aggregator.py]: yes. I just rebased on that.

File: README.md [hunk 2]
```diff
@@ -408,6 +408,28 @@ instance (without any instrumentation)
   through variable `driver`.
 
 
+### Managing requirements
+
+We use [`pip-tools`](https://github.com/jazzband/pip-tools) to pin
```
[the remark this answers is not in the document]
Comment (englehardt) [reply README.md]: There's a bit of discussion around this here: https://github.com/jazzband/pip-tools/issues/398. I think it's worth keeping despite the minor complexity for the developer handling the dependency upgrades. IMO it's better to pay that complexity cost at the the time of adding new dependencies than to pay the cost associated with chasing down bugs that result from an inconsistent environment that changes depending on whether the dev requirements were installed.

As an example, let's say `requirements.txt` installs the newest version of `pyarrow`. This version causes the bug we experienced in #498. But some dependency in `requirements-dev.txt` pins to an older version of the library (which doesn't cause #498 ). If we run `pip install -r requirements.txt -r requirements-dev.txt` pip will throw an error about conflicting and incompatible pinned versions -- this is actually what prompted me to look in to layering. If we instead run `pip install -r requirements.txt` followed by `pip install -r requirements-dev.txt`, `pip` will first install the new version of `pyarrow` and then uninstall it and install the older pinned version from `requirements-dev.txt`. All of the CI tests will pass just fine, but any user who only runs `pip install -r requirements.txt` will receive the newer version of `pyarrow` and thus experience the bug.

[This post](https://hynek.me/articles/python-app-deps-2018/) discusses other tools that try to hide some of this complexity, but ultimately recommends `pip-tools` due to constant issues.

File: requirements.in [hunk 3]
```diff
@@ -0,0 +1,29 @@
+beautifulsoup4
```
[the remark this answers is not in the document]
Comment (englehardt) [reply requirements.in]: Yes, feeding an empty `.in` file will result in an empty `.txt` file.

Think of the `requirements.txt` files as compiled versions of the `requirement.in` files. The `.in` files specify the packages we want to use, and any pins we want to manually specify. Then, the `.txt` file includes the full set of pinned dependencies (including dependencies of dependencies). So we will need to regularly re-compile the `.txt` files from the `.in` files to have the latest versions of all of the packages we rely on, but know that between updates we will have a deterministic set of dependencies.

**Comment by motin** [comment]
I'd love to see a move to anaconda eventually (a vital step towards https://github.com/mozilla/OpenWPM/issues/503), and this could be a good moment to do it. Anaconda's `environment.yml` is a well known standard, involves no pre-processing step and `conda env export` pins versions by default. There is a test-passing PR for this in https://github.com/mozilla/OpenWPM/pull/429 - it needs rebasing and cleanup for sure, but it may be worth going down that path now rather than first introducing this and then moving to conda.

File: requirements.in (hunk 3 above)
[the remark this answers is not in the document]
Comment (nhnt11) [reply requirements.in]: Gotcha, sounds like package.json and package-lock.json in the world of Node.js

**Comment by englehardt** [comment]
> However, I'd love to see a move to anaconda eventually (a vital step towards #503), and this could be a good moment in time to do it. Anaconda's environment.yml is a well known standard, involves no pre-processing step and conda env export pins versions by default. There is a test-passing PR for this in #429 - it needs rebasing and cleanup for sure, but it may be worth going down that path now rather than first introducing this and then moving to conda.

Thanks! I still need to spend some time to think through Anaconda and whether it's something we'd want to move to. For now, this PR solves an immediate need and I don't think we should block on making a decision about #429.

**Pull request merged**